test/expected/sql_query_results_optimized.out

```diff
@@ -369,19 +369,20 @@ FROM hyper_1_tz GROUP BY t ORDER BY t DESC limit 2;
 
 EXPLAIN (costs off) SELECT time_bucket(10, time) t, avg(series_0), min(series_1), avg(series_2)
 FROM hyper_1_int GROUP BY t ORDER BY t DESC limit 2;
-                                          QUERY PLAN                                          
-----------------------------------------------------------------------------------------------
+                                                                        QUERY PLAN                                                                        
+----------------------------------------------------------------------------------------------------------------------------------------------------------
  Limit
    ->  GroupAggregate
-         Group Key: (((hyper_1_int."time" / 10) * 10))
-         ->  Result
-               ->  Merge Append
-                     Sort Key: (((hyper_1_int."time" / 10) * 10)) DESC
-                     ->  Index Scan using time_plain_int on hyper_1_int
-                     ->  Index Scan using _hyper_3_3_chunk_time_plain_int on _hyper_3_3_chunk
-                     ->  Index Scan using _hyper_3_4_chunk_time_plain_int on _hyper_3_4_chunk
-                     ->  Index Scan using _hyper_3_5_chunk_time_plain_int on _hyper_3_5_chunk
-(10 rows)
+         Group Key: (CASE WHEN (hyper_1_int."time" >= 0) THEN ((hyper_1_int."time" / 10) * 10) ELSE (((hyper_1_int."time" / 10) - 1) * 10) END)
```

Comment:
This output is quite jarring. Can't we hide this with a native function? Is native code really slower than inlined SQL?

Reply:
Performance of inlined SQL function and C function is about the same. I will turn the int time_bucket variants into c functions.